pkg/controller/nstemplatetier/nstemplatetier_controller.go

```diff
@@ -301,6 +305,14 @@ func (r *NSTemplateTierReconciler) activeTemplateUpdateRequests(logger logr.Logg
 	return count, false, nil
 }
 
+// maxUpdateFailuresReached checks if the number of failure to update the MasterUserRecord is beyond the configured threshold
+func maxUpdateFailuresReached(tur toolchainv1alpha1.TemplateUpdateRequest, threshod int) bool {
+	return condition.Count(tur.Status.Conditions,
+		toolchainv1alpha1.TemplateUpdateRequestComplete,
+		corev1.ConditionFalse,
+		toolchainv1alpha1.TemplateUpdateRequestUnableToUpdateReason) > threshod
```

Comment:
Should it be `>=`?
Otherwise we will actually make `maxAttemps` + 1 attempts, won't we?
If I'm not missing something I would set the default maxAttemps to 2 and use `>=` here.

Reply:
hmm, I thought my tests proved that the comparison was correct. 
Maybe the confusion is that the threshold is the number of failed attempts that allow for another retrial. So if `threshold=1`, then the controller may fail once, try again and then give up.
I understand it can be confusing, so I'm fine with changing the value to `2` and using `>=` instead. Just let me know ;)

Reply:
Yes, exactly. You use this as a failure threshold. But IMO it can be confused with overall number of update attempts.
And I'm fine to keep it as a failure treshold but maybe we can rename it and add a comment to avoid confusion.
Something like:

```
// GetMasterUserRecordUpdateFailureThreshold returns the failure threshold for MUR updates. When a MUR update fails, we will re-try N times before giving up and stopping the entire tier update process.
func (c *Config) GetMasterUserRecordUpdateFailureThreshold() int {
	return c.host.GetInt(varMasterUserRecordUpdateFailureThreshold)
}
```

Reply:
But this is minor.

Reply:
done in b8cc04b